rotkehlchen/serialization/schemas.py

```diff
@@ -1,100 +1,236 @@
+            expected_asset_type: Optional[AssetType] = None,
+            coingecko: Optional['Coingecko'] = None,
+            cryptocompare: Optional['Cryptocompare'] = None,
+    ) -> None:
+        super().__init__()
+        self.identifier_required = identifier_required
+        self.expected_asset_type = expected_asset_type
+        self.coingecko_obj = coingecko
+        self.cryptocompare_obj = cryptocompare
+
+    @validates_schema
+    def validate_schema(
+            self,
+            data: dict[str, Any],
+            **_kwargs: Any,
+    ) -> None:
+        if self.expected_asset_type is None and data['asset_type'] is None:
+            raise ValidationError(message='Asset schema asset_type should be given', field_name='asset_type')  # noqa: E501
+        if self.expected_asset_type is not None and data['asset_type'] is not None and data['asset_type'] != self.expected_asset_type:  # noqa: E501
+            raise ValidationError(message=f'Asset schema asset_type should be {self.expected_asset_type}', field_name='asset_type')  # noqa: E501
+        if self.identifier_required is True and data['identifier'] is None:
+            raise ValidationError(message='Asset schema identifier should be given', field_name='identifier')  # noqa: E501
+        if self.coingecko_obj is not None:
+            _validate_single_oracle_id(data, 'coingecko', self.coingecko_obj)
+        if self.cryptocompare_obj is not None:
+            _validate_single_oracle_id(data, 'cryptocompare', self.cryptocompare_obj)
+
+    @post_load
+    def transform_data(self, data: dict[str, Any], **_kwargs: Any) -> dict[str, CryptoAsset]:
+        crypto_asset = CryptoAsset.initialize(
+            # asset id needs to be unique but no combination of asset data is guaranteed to be
+            # unique. And especially with the ability to edit assets we need an external uuid.
+            identifier=data['identifier'] if data['identifier'] is not None else str(uuid4()),
+            name=data['name'],
+            symbol=data['symbol'],
+            asset_type=self.expected_asset_type if self.expected_asset_type is not None else data['asset_type'],  # noqa: E501
+            started=data['started'],
+            swapped_for=data['swapped_for'],
+            forked=data['forked'],
+            coingecko=data['coingecko'],
+            cryptocompare=data['cryptocompare'],
+        )
+        return {'crypto_asset': crypto_asset}
+
+
+class EvmTokenSchema(BaseCryptoAssetSchema, RequiredEvmAddressSchema):
+    token_kind = SerializableEnumField(enum_class=EvmTokenKind, required=True)
+    decimals = fields.Integer(
+        strict=True,
+        validate=webargs.validate.Range(
+            min=0,
+            max=18,
+            error='Evm token decimals should range from 0 to 18',
+        ),
+        required=True,
+    )
+    protocol = fields.String(load_default=None)
+    underlying_tokens = fields.List(fields.Nested(UnderlyingTokenInfoSchema), load_default=None)
```

Comment:
pseudocode - did not test

```suggestion
    underlying_tokens = DelimitedOrNormalLIst(fields.Nested(UnderlyingTokenInfoSchema), load_default=None)
```